src/common/datomish/pair_chan.cljc

```diff
@@ -71,6 +71,13 @@
      (consume-pair (cljs.core.async/<! ~pc-chan))
      (consume-pair (clojure.core.async/<! ~pc-chan))))
 
+(defmacro <??
+  "Takes from the channel if it's non-nil."
+  [pc-chan]
+  `(let [c# ~pc-chan]
+     (when c#
+       (datomish.pair-chan/<? c#))))
```

Comment:
That code _expands_. Assume the input bound to `pc-chan` is an expression, say `(foo 5)`, that returns `nil` when eventually run.

It expands at compile time to:

```
(let [c123 (foo 5)]
  (when c123
    (<? c123)))
```

When this code runs, inside of a `go-pair` block, it does the right thing: that is, if `(foo 5)` returns a value (presumably a channel), take from it; if it returns `nil` or `false`, do nothing.